solver/src/main/java/org/chocosolver/solver/propagation/PropagationEngine.java

```diff
@@ -189,6 +189,9 @@ public void propagate() throws ContradictionException {
             manageModifications();
             for (int i = nextNotEmpty(); i > -1; i = nextNotEmpty()) {
                 assert !pro_queue[i].isEmpty() : "try to pop a propagator from an empty queue";
+                if (model.getSolver().isTimeLimitMet()) {
```

Comment:
why not calling model.getSolver().isStopCriterionMet() ? 
This would avoid modifying the Solver class.

Reply:
The idea was to have a simple check on the time limit, and not check all the stopping criterion before the propagation of each propagator.
Indeed, all stopping criterion but the time limit do not depend on the problem's state outside of fixpoint. Unless you can think of one that would.

Reply:
Personnaly, I would have move the code to the end of the while-loop just to keep the main purpose (ie, propagating) clear